src/core/net/ip6.cpp

```diff
@@ -1068,6 +1069,22 @@ Error Ip6::SendRaw(OwnedPtr<Message> aMessagePtr)
 
     if (header.GetDestination().IsMulticast())
     {
+#if OPENTHREAD_CONFIG_FILTER_LOCAL_ORIGINATED_MULTICAST_PACKETS
+        // For multicast packets originated from this device, only forward to thread
+        // network if destination is in the multicast listener table
+        if (!aMessagePtr->IsOriginThreadNetif() && Get<ThreadNetif>().HasUnicastAddress(header.GetSource()))
```

Comment:
> `!aMessagePtr->IsOriginThreadNetif()`

Shall we check `aMessagePtr->IsOriginHostUntrusted()` instead? When `IsOriginThreadNetif()` is false, it could be `IsOriginHostTrusted()` which is also from OT stack.

Reply:
Thanks, it should indeed by untrusted source that we are filtering. 👍